sysconfig/cloudinit.go

```diff
@@ -208,6 +210,11 @@ const (
 // cloud-init may be doing something and will return CloudInitEnabled when we
 // do not recognize the state returned by the cloud-init status command.
 func CloudInitStatus() (CloudInitState, error) {
+	lp, err := exec.LookPath("cloud-init")
+	if err != nil {
+		return CloudInitNotFound, nil
+	}
```

Comment:
I think this check should happen after the restrict file checking happens, so that if we do end up restricting cloud-init, it will show up as "cloud-init already restricted/disabled" etc.